client/components/tinymce/plugins/wpcom-track-paste/plugin.js

```diff
@@ -46,39 +42,38 @@ function pasteTrack( editor ) {
 		return SOURCE_UNKNOWN;
 	}
 
-	function flattenArray( orgTypes ) {
-		// As per https://html.spec.whatwg.org/multipage/interaction.html#datatransfer
-		// types should be an array containing the available formats that were set on copy.
-		// Some browsers, as Firefox, return a DOMStringList instead, see
-		// https://developer.mozilla.org/en-US/docs/Web/API/DOMStringList
-		let types = [];
-		if ( orgTypes.contains ) {
-			types = Array.from( orgTypes );
-		} else {
-			types = orgTypes.slice();
-		}
-		return types.reduce( ( a, b ) => {
-			return a + ', ' + b;
-		} );
+	function toString( types ) {
+		// types should be an array. Some browsers, as Firefox, return a DOMStringList instead.
+		// We need coverage for thoses cases, hence the Array.from conversion.
+		//
+		// https://html.spec.whatwg.org/multipage/interaction.html#datatransfer
+		// DOMStringListhttps://developer.mozilla.org/en-US/docs/Web/API/DOMStringList
+		return Array.from( types ).join( ', ' );
 	}
 
-	function onPasteEditor( event ) {
-		recordEvent( flattenArray( event.clipboardData.types ), 'tinymce' );
+	function onPasteFromTinyMCEEditor( event ) {
+		recordEvent( 'tinymce-editor', toString( event.clipboardData.types ) );
 	}
 
-	function onPasteTextArea( event ) {
-		recordEvent( flattenArray( event.clipboardData.types ), 'html' );
+	function onPasteFromHTMLEditor( event ) {
+		recordEvent( 'html-editor', toString( event.clipboardData.types ) );
```

Comment:
if we call `toString` on every call to `recordEvent` do we even need to call `toString` here? why not move it into `recordEvent` since `recordEvent` is already tightly coupled to recording these specific types of events.

could even make that clear by renaming to `recordPasteEvent` or something like that